packages/web/src/features/languages/ImportLanguageGlossesView.tsx

```diff
@@ -32,29 +25,89 @@ export async function importLanguageGlossesLoader({
     params.code ?? 'unknown'
   );
   const importLanguages = await apiClient.import.getLanguages();
-  return { language, importLanguages };
+  let currentJob;
+  try {
+    currentJob = await apiClient.languages.getImportStatus(
+      params.code ?? 'unknown'
+    );
+  } catch (error) {
+    if (!(error instanceof ApiClientError) || error.status !== 404) {
+      throw error;
+    }
+  }
+  return { language, importLanguages, currentJob };
 }
 
 interface FormData {
   import: string;
 }
 
 export default function ImportLanguageGlossesView() {
-  const { language, importLanguages } = useLoaderData() as {
-    language: GetLanguageResponseBody;
-    importLanguages: GetLanguageImportOptionsResponseBody;
-  };
+  const { language, importLanguages, currentJob } = useLoaderData() as Awaited<
+    ReturnType<typeof importLanguageGlossesLoader>
+  >;
   const flash = useFlash();
   const { t } = useTranslation();
   const confirmationDialog = useRef<ConfirmationDialogRef>(null);
 
   const formContext = useForm<FormData>();
+
+  const [importStatus, setImportStatus] = useState<
+    'idle' | 'running' | 'complete' | 'error'
+  >(() => {
+    if (currentJob) {
+      if (currentJob.endDate) {
+        if (
+          Date.now() - new Date(currentJob.endDate).valueOf() <
+          10 * 60 * 1000 // 10 minutes
```

Comment:
Why check for 10 minutes? Why not use the `succeeded` field?

Reply:
If you only use the succeeded flag, then all subsequent views of this page will show the success step. This assumes that if you revisit the page in ten minutes, that you are wanting to start a new import. Maybe that should be 24 hours. It should though determine the status within that 10 minutes based on the succeeded field. I'll fix that